### ISSUE TITLE
Housing Price Prediction

### ISSUE THREAD

**Issue by rakki-18** [WebClub-NITK/Hacktoberfest-2k20#28]
### Description
Given the features of the house, predict the price of the house using any of the regression algorithms .
You can download the dataset from [here](https://www.cs.toronto.edu/~delve/data/boston/bostonDetail.html)
or you could load it by typing
```
from sklearn.datasets import load_boston
data=load_boston()
```

### Details
- Technical Specifications:  python, numpy, pandas, scikit-learn
- Type of issue: Multiple
- Time Limit: 2 days
### Issue requirements / progress
Train your model on the training data  and predict the values using the test data and use any metric to evaluate the performance.
#### Note: Each PR should be a solution using only one model
- [x] Implementing a linear regression model
- [x]  Implementing a polynomial regression model
- [x] Implementing a lasso regression model
- [x] Implementing a ridge regression model

### Resources
https://towardsdatascience.com/linear-regression-on-boston-housing-dataset-f409b7e4a155
https://towardsdatascience.com/polynomial-regression-bbe8b9d97491

### Directory Structure
Make a folder called "28_Housing_Price" in the "MachineLearing" folder if it doesn't exist yet.
Place your solution file in 
- For linear regression
```
/MachineLearning/28_Housing_Price/linearRegression/<your_file>
```
- For polynomial regression
```
/MachineLearning/28_Housing_Price/polynomialRegression/<your_file>
```
- For lasso regression
```
/MachineLearning/28_Housing_Price/lassoRegression/<your_file>
```
- For ridge regression
```
/MachineLearning/28_Housing_Price/ridgeRegression/<your_file>
```
  

#### Note
1. Please claim the issue first by commenting here along with the model you are going to be using  before starting to work on it.
2. Once you are done with the task and have created a Pull Request, please tag @rakki-18 to request a review.
Happy learning!

**Comment by Hemant7499** [comment]
I would like to work on this using the linear regression model.

**Comment by rakki-18** [comment]
@Smally12345 Sure, go ahead.

**Comment by itsanshulverma** [comment]
@rakki-18 I would like to work on polynomial regression model

**Comment by shaonakundu** [comment]
@rakki-18 I would like to work on linear regression model, I have knowledge in this domain so please allow me to contribute this task..can you assign me for linear regression?

**Comment by RohanSahana** [comment]
@rakki-18  I would like to use the Multiple Linear Regression.

**Comment by rakki-18** [comment]
> @rakki-18 I would like to work on polynomial regression model

@itsanshulverma Go ahead but please stick to the time limit

**Comment by itsanshulverma** [comment]
@rakki-18  I would like to work on lasso regression and ridge regression models too. Please assign it to me.

**Comment by itsanshulverma** [comment]
> > @rakki-18 I would like to work on polynomial regression model
> 
> @itsanshulverma Go ahead but please stick to the time limit

Done! Please review PR.

**Comment by itsanshulverma** [comment]
> @rakki-18 I would like to work on lasso regression and ridge regression models too. Please assign it to me.

The last two models done! Please review the PRs.

**Comment by rakki-18** [comment]
@itsanshulverma all the code looks fine.
But the directory structure is wrong. You were supposed to make a folder named "28_Housing_Price" in the "MachineLearning" folder and inside the "28_Housing_Price" folder, you had to make a folder "name_of_model" and inside that,  you had to put all the relevant files.
Please resolve this and send another PR

**Comment by itsanshulverma** [comment]
> @itsanshulverma all the code looks fine.
> But the directory structure is wrong. You were supposed to make a folder named "28_Housing_Price" in the "MachineLearning" folder and inside the "28_Housing_Price" folder, you had to make a folder "name_of_model" and inside that,  you had to put all the relevant files.
> Please resolve this and send another PR

I have done the same.
I have created a separate branch for each model, with the same directory structure.

Here are the links:
- [Polynomial Regression](https://github.com/itsanshulverma/Hacktoberfest-2k20/tree/polynomial_regression/MachineLearning/28_Housing_Price/polynomialRegression)
- [Lasso Regression](https://github.com/itsanshulverma/Hacktoberfest-2k20/tree/lasso_regression/MachineLearning/28_Housing_Price/lassoRegression)
- [Ridge Regression](https://github.com/itsanshulverma/Hacktoberfest-2k20/tree/ridge_regression/MachineLearning/28_Housing_Price/ridgeRegression)

I am not sure what to do next. @rakki-18 Please guide. Thanks in advance.

**Comment by rakki-18** [comment]
> > @itsanshulverma all the code looks fine.
> > But the directory structure is wrong. You were supposed to make a folder named "28_Housing_Price" in the "MachineLearning" folder and inside the "28_Housing_Price" folder, you had to make a folder "name_of_model" and inside that,  you had to put all the relevant files.
> > Please resolve this and send another PR
> 
> I have done the same.
> I have created a separate branch for each model, with the same directory structure.
> 
> Here are the links:
> 
> * [Polynomial Regression](https://github.com/itsanshulverma/Hacktoberfest-2k20/tree/polynomial_regression/MachineLearning/28_Housing_Price/polynomialRegression)
> * [Lasso Regression](https://github.com/itsanshulverma/Hacktoberfest-2k20/tree/lasso_regression/MachineLearning/28_Housing_Price/lassoRegression)
> * [Ridge Regression](https://github.com/itsanshulverma/Hacktoberfest-2k20/tree/ridge_regression/MachineLearning/28_Housing_Price/ridgeRegression)
> 
> I am not sure what to do next. Please guide. Thanks in advance.

The folder under "MachineLearning" should be "28_Housing_Price" and not "28_Housing_Price/polynomialRegression".
In the "28_Housing_Price" folder, make a folder named "polynomialRegression" and inside that put all your relevant files.
Do the same for all the other 3 models and make a separate PR for the 3 of them.

**Comment by rakki-18** [comment]
@itsanshulverma Everything is fine. Good work

**Comment by itsanshulverma** [comment]
> @itsanshulverma Everything is fine. Good work

Great! Happy to contribute! Thank you!
So, when will the PRs be merged?

@rakki-18
